src/python/grpcio_tests/tests_aio/unit/call_test.py

```diff
@@ -11,186 +11,288 @@
+                messages_pb2.SimpleRequest(), timeout=timeout)
+
+            with self.assertRaises(grpc.RpcError) as exception_context:
+                await call
+
+            self.assertTrue(call.done())
+            self.assertEqual(await call.code(),
+                             grpc.StatusCode.DEADLINE_EXCEEDED)
+
+            # Exception is cached at call object level, reentrance
+            # returns again the same exception
+            with self.assertRaises(grpc.RpcError) as exception_context_retry:
+                await call
+
+            self.assertIs(exception_context.exception,
+                          exception_context_retry.exception)
+
+    async def test_call_code_awaitable(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            hi = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = hi(messages_pb2.SimpleRequest())
+            self.assertEqual(await call.code(), grpc.StatusCode.OK)
+
+    async def test_call_details_awaitable(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            hi = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = hi(messages_pb2.SimpleRequest())
+            self.assertEqual('', await call.details())
+
+    async def test_cancel_unary_unary(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            hi = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = hi(messages_pb2.SimpleRequest())
+
+            self.assertFalse(call.cancelled())
+
+            # TODO(https://github.com/grpc/grpc/issues/20869) remove sleep.
+            # Force the loop to execute the RPC task.
+            await asyncio.sleep(0)
+
+            self.assertTrue(call.cancel())
+            self.assertFalse(call.cancel())
+
+            with self.assertRaises(asyncio.CancelledError) as exception_context:
+                await call
+
+            self.assertTrue(call.cancelled())
+            self.assertEqual(await call.code(), grpc.StatusCode.CANCELLED)
+            self.assertEqual(await call.details(),
+                             'Locally cancelled by application!')
+
+            # NOTE(lidiz) The CancelledError is almost always re-created,
+            # so we might not want to use it to transmit data.
+            # https://github.com/python/cpython/blob/master/Lib/asyncio/tasks.py#L785
```

Comment:
Oks, so we follow the same criteria and we never save a CancelledError exception.